src/lazy/expanded/sequence.rs

```diff
@@ -1,16 +1,70 @@
+use bumpalo::collections::Vec as BumpVec;
+
+use crate::element::iterators::SymbolsIterator;
 use crate::lazy::decoder::{LazyDecoder, LazyRawSequence, LazyRawValueExpr, RawValueExpr};
-use crate::lazy::expanded::macro_evaluator::{TransientEExpEvaluator, TransientTdlMacroEvaluator};
-use crate::lazy::expanded::template::TemplateSequenceIterator;
+use crate::lazy::expanded::macro_evaluator::{MacroEvaluator, RawEExpression, ValueExpr};
+use crate::lazy::expanded::template::{
+    AnnotationsRange, ExprRange, TemplateMacroRef, TemplateSequenceIterator,
+};
 use crate::lazy::expanded::{
     EncodingContext, ExpandedAnnotationsIterator, ExpandedAnnotationsSource, ExpandedValueSource,
     LazyExpandedValue,
 };
-use crate::{Annotations, IonResult, IonType, Sequence};
+use crate::result::IonFailure;
+use crate::{IonError, IonResult, IonType};
+
+#[derive(Copy, Clone, Debug)]
+pub struct Environment<'top, 'data, D: LazyDecoder<'data>> {
+    expressions: &'top [ValueExpr<'top, 'data, D>],
+}
+
+pub const fn empty_args<'top, 'data, D: LazyDecoder<'data>>() -> &'top [ValueExpr<'top, 'data, D>] {
+    &[]
+}
```

Comment:
For anyone curious: `const` and `static` variables cannot use generics from their containing function signature.

Since this was the only usage of `empty_args()`, I made it a nested function for easier reading / improved locality.

Reply:
Turns out I was able to simplify this to just
```rust
    pub fn empty() -> Environment<'top, D> {
        Environment { expressions: &[] }
    }
```
I was sure I needed a `static` or `const` to prevent the `[]` from living on the stack, but I guess not!